src/core/public/http/http_service.test.ts

```diff
@@ -313,6 +313,153 @@ describe('http requests', () => {
   });
 });
 
+describe('interception', () => {
+  const tap = (handler: Function) => (value: any) => {
+    handler();
+    return value;
+  };
+  const { http } = setup();
+
+  beforeEach(() => {
+    fetchMock.get('*', { foo: 'bar' });
+  });
+
+  afterEach(() => {
+    fetchMock.restore();
+    http.removeAllInterceptors();
```

Comment:
question: are there any use cases in our code except testing? I wouldn't add any api for tests only. we can re-create http service in `beforeEach` to cleanup state

Reply:
Yes, this is used when stopping, and any other consumers that may want to clear such as security.